sklearn/metrics/scorer.py

```diff
@@ -282,6 +282,91 @@ def check_scoring(estimator, scoring=None, allow_none=False):
             "have a 'score' method. The estimator %r does not." % estimator)
 
 
+def _check_multimetric_scoring(estimator, scoring=None, allow_none=False):
+    """Check the scoring parameter in cases when multiple metrics are allowed
+
+    Parameters
+    ----------
+
+    estimator : sklearn estimator instance
+        The estimator for which the scoring will be applied.
+
+    scoring : string, callable or None, optional, default: None
+        A single string (see :ref:`_scoring_parameter`) or a callable
+        (see :ref:`_scoring`) to evaluate the predictions on the test set.
+
+        For evaluating multiple metrics either give a list of unique strings
+        or a dict with names as keys and callables as values.
+
+        NOTE that when using custom scorers, each scorer should return a single
+        value. Single scorers returning a list/array of values may be wrapped
+        into multiple scorers that return one value each.
+
+        If ``None``, the estimator's default scorer, is used.
+        The return value in that case will be ``{'score': <default_scorer>}``.
+        If the estimator's default scorer is not available, a ``TypeError``
+        is raised.
+
+    allow_none : boolean, default: False
+        Whether to allow None even if estimator does not have a score method.
+
+    Returns
+    -------
+
+    scorers_dict : dict
+        A dict mapping each scorer name to its validated scorer.
+
+        The validated scorer could be None if ``allow_none`` is ``True``
+        (``{'score': None}``).
+
+        The scorer_name is set to ``'score'`` if the default scorer is
+        used.
+
+    is_multimetric : bool
+        True if scorer is a list/tuple or dict of callables
+        False if scorer is None/str/callable
+    """
+    if callable(scoring) or scoring is None or isinstance(scoring,
+                                                          six.string_types):
+        scorers = {"score": check_scoring(estimator, scoring=scoring,
+                                          allow_none=allow_none)}
+        return scorers, False
+    else:
+        err_msg_generic = ("scoring should either be a single string or "
```

Comment:
I think we should have more precise error messages. If someone gave a sequence of anything but strings, we should tell them to use a dict, and if they have a sequence with non-unique strings, we should just tell them that they should remove those.